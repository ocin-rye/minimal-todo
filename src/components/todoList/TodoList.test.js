import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TodoList from './index';

describe('<TodoList />', () => {
  let todoList;

  const testTodoList = [
    {
      completed: false,
      id: 'testTodoId',
      text: 'Test todo text.',
    },
    {
      completed: true,
      id: 'SGcqcq0TbTPqvyWOMIKtZ',
      text: 'Duis maximus lacinia pellentesque.',
    },
    {
      completed: false,
      id: 'AHdfoOWOYnObQQH4zWcdL',
      text:
        'Aenean molestie elit at dictum finibus. Vestibulum porttitor at lorem viverra accumsan.',
    },
  ];

  const testTodoListEmpty = [];

  it('renders initial state without crashing', () => {
    let todoListFilter = 'All';
    todoList = shallow(<TodoList todoList={testTodoList} todoListFilter={todoListFilter} />);

    expect(toJson(todoList)).toMatchSnapshot();
  });

  it('renders updated todo list', () => {
    let todoListFilter = 'All';
    todoList = shallow(<TodoList todoList={testTodoList} todoListFilter={todoListFilter} />);

    todoList.setProps({ todoList: [] });

    todoList.update();

    expect(todoList.state().todoList).toEqual([]);
  });

  describe('renders Header', () => {
    it('renders empty array todo list header', () => {
      let todoListFilter = 'All';
      todoList = shallow(<TodoList todoList={testTodoListEmpty} todoListFilter={todoListFilter} />);

      expect(todoList.find('.header').text()).toEqual('No Todos');
    });

    it('renders empty array active todo list header', () => {
      let todoListFilter = 'Active';
      todoList = shallow(<TodoList todoList={testTodoListEmpty} todoListFilter={todoListFilter} />);

      expect(todoList.find('.header').text()).toEqual('No Active Todos');
    });

    it('renders empty array completed todo list header', () => {
      let todoListFilter = 'Completed';
      todoList = shallow(<TodoList todoList={testTodoListEmpty} todoListFilter={todoListFilter} />);

      expect(todoList.find('.header').text()).toEqual('No Completed Todos');
    });

    it('renders populated todo list header', () => {
      let todoListFilter = 'All';
      todoList = shallow(<TodoList todoList={testTodoList} todoListFilter={todoListFilter} />);

      expect(todoList.find('.header').text()).toEqual('All Todos: 3');
    });

    it('renders populated active todo list header', () => {
      let todoListFilter = 'Active';
      todoList = shallow(<TodoList todoList={testTodoList} todoListFilter={todoListFilter} />);

      expect(todoList.find('.header').text()).toEqual('Active Todos: 2');
    });

    it('renders populated completed todo list header', () => {
      let todoListFilter = 'Completed';
      todoList = shallow(<TodoList todoList={testTodoList} todoListFilter={todoListFilter} />);

      expect(todoList.find('.header').text()).toEqual('Completed Todos: 1');
    });
  });

  describe('handles input submit', () => {});
});
