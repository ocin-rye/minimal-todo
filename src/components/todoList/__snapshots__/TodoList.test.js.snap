// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> renders initial state without crashing 1`] = `
<div
  className="container"
>
  <p
    className="textCenter header"
  >
    All Todos: 3
  </p>
  <Todo
    key="testTodoId"
    todo={
      Object {
        "completed": false,
        "id": "testTodoId",
        "text": "Test todo text.",
      }
    }
  />
  <Todo
    key="SGcqcq0TbTPqvyWOMIKtZ"
    todo={
      Object {
        "completed": true,
        "id": "SGcqcq0TbTPqvyWOMIKtZ",
        "text": "Duis maximus lacinia pellentesque.",
      }
    }
  />
  <Todo
    key="AHdfoOWOYnObQQH4zWcdL"
    todo={
      Object {
        "completed": false,
        "id": "AHdfoOWOYnObQQH4zWcdL",
        "text": "Aenean molestie elit at dictum finibus. Vestibulum porttitor at lorem viverra accumsan.",
      }
    }
  />
</div>
`;
